File: bestia-models/src/main/java/net/bestia/model/domain/StatusPoints.java
package net.bestia.model.domain;

import com.fasterxml.jackson.annotation.JsonProperty;

public interface StatusPoints {

	/**
	 * Defense against regular physical attacks.
	 * 
	 * @return Physical attack defense.
	 */
	@JsonProperty("def")
	int getDefense();

	/**
	 * Returns the magic defense.
	 * 
	 * @return The magic defense.
	 */

	@JsonProperty("mdef")
	int getMagicDefense();

	/**
	 * Returns the physical strength.
	 * 
	 * @return Strength
	 */
	@JsonProperty("str")
	int getStrength();

	/**
	 * Returns the vitality.
	 * 
	 * @return Vitality.
	 */
	@JsonProperty("vit")
	int getVitality();

	/**
	 * Returns the intelligence.
	 * 
	 * @return Intelligence.
	 */
	@JsonProperty("int")
	int getIntelligence();

	/**
	 * Returns the agility of the entity.
	 * 
	 * @return The agility.
	 */
	@JsonProperty("agi")
	int getAgility();

	@JsonProperty("will")
	int getWillpower();

	@JsonProperty("dex")
	int getDexterity();

	void setDexterity(int dexterity);

	void setWillpower(int willpower);

	void setAgility(int agi);

	void setIntelligence(int intel);

	void setVitality(int vit);

	void setStrenght(int str);

	void setMagicDefense(int mdef);

	void setDefense(int def);

}

File: bestia-models/src/main/kotlin/net/bestia/model/bestia/StatusPoints.kt
package net.bestia.model.bestia

import com.fasterxml.jackson.annotation.JsonProperty

interface StatusPoints {
  @get:JsonProperty("def")
  var defense: Int

  @get:JsonProperty("mdef")
  var magicDefense: Int

  @get:JsonProperty("str")
  var strength: Int

  @get:JsonProperty("vit")
  var vitality: Int

  @get:JsonProperty("int")
  var intelligence: Int

  @get:JsonProperty("agi")
  var agility: Int

  @get:JsonProperty("will")
  var willpower: Int

  @get:JsonProperty("dex")
  var dexterity: Int

  /**
   * Copies all values from the given [StatusPoints] instance.
   *
   * @param rhs The [StatusPoints] object to copy.
   */
  fun set(rhs: StatusPoints)
}